### PR TITLE
Add restrictions to app store extension tasks

### PR DESCRIPTION
**Pull request by DaniilShmelev**
**Task name**: AppStoreRelease, AppStorePromote, IpaResign

**Documentation changes required:** No

**Added unit tests:** No

**Checklist**:
- [x] Checked that applied changes work as expected:

AppStoreRelease:
[service connection basic production no fastlane install](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2702&view=results)
[service connection api key testFlight no fastlane install](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2706&view=results)
[api key auth production](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2707&view=results)
[basic auth testFlight](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2731&view=results)
[skip 2fa](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2709&view=results)

IpaResign:
[using file](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2724&view=results)
[using signing identity](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2726&view=results)

AppStorePromote:
[api key auth](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2727&view=results)
[basic auth](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2728&view=results)
[service connection basic](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2729&view=results)
[service connection api key](https://abtttestorg.visualstudio.com/XcodeBuildTest/_build/results?buildId=2730&view=results)

### CODE REVIEW

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks! Let's keep it until we will be able to get telemetry for this feature/have agent knob to switch it on/off

File: Tasks/AppStorePromote/task.json [hunk 1]
```diff
@@ -13,10 +13,10 @@
     "demands": [ "xcode" ],
     "version": {
         "Major": "1",
-        "Minor": "183",
+        "Minor": "184",
         "Patch": "0"
     },
-    "minimumAgentVersion": "1.95.3",
+    "minimumAgentVersion": "2.181.1",
```
Comment (EzzhevNikita) [inline Tasks/AppStorePromote/task.json:19]: Agent version should be 2.182.1

**Review by EzzhevNikita — APPROVED** [review]
LGTM! But please take a look at the comment

File: Tasks/AppStorePromote/task.json (hunk 1 above)
Comment (DaniilShmelev) [reply Tasks/AppStorePromote/task.json]: Thanks, fixed!

**Pull request merged**